src/vte.c

```diff
@@ -675,6 +682,45 @@ const gchar *vte_get_working_directory(void)
 				g_free(cwd);
 			}
 		}
+		else
+		{
+			gint status = 0;
+			gchar *stdout = NULL;
+			gchar *pid_str = g_strdup_printf("%d", pid);
+			gchar *argv[] = {"lsof", "-a", "-d", "cwd", "-F", "n", "-p", pid_str, NULL};
```

Comment:
Ideally you'd make this a `const gchar*` array, so that literals can be constant; and in the call below just cast up to `gchar**`.

Reply:
`utils_spawn_sync` calls `spawn_sync` which on Linux eventually calls `g_spawn_async` which specifies the array as `gchar**`, not a `const` in sight, so it may be a pain to try to `const`ize this.

Reply:
Note the `exec` family of system calls takes `char* const argv[]` so `g_spawn` is looser than needed.